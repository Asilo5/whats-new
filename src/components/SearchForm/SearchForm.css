
.search-button {
  background: #B56AF0;
  font-size: 17px;
  padding: 14px;
  border: none;
  border-radius: 22px;
  margin: 13px;
  color: white;
   width: 10em;
}

.search-input{
  font-size: 25px;
  padding: 5px;
  border: none;
  border-radius: 5px 0px 0px 5px;
  width: 82%;
  background-color: transparent;
  }