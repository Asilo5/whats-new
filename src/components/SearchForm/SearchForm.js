import React, { Component } from 'react';
import './SearchForm.css';


class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
        headline: ''
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name] : e.target.value })
  }

  findArticle = (e) => {
    e.preventDefault();

    this.props.findNews(e, this.state.headline);

    this.setState({
        headline: ''
    })
  }

  render () {
    return (
      <form>
        <input 
          className='search-input'
          type='text' 
          name='headline'
          placeholder='Search for news article here.'
          value={this.state.headline}
          onChange={e => this.handleChange(e)}
        />
        <button className='search-button' onClick={this.findArticle}> Search Now </button>
      </form>
    )
  }
}

export default SearchForm;