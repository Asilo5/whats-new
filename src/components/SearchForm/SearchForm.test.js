import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
    let wrapper;
    const mockFindArticle = jest.fn();
    const mockHeadline = 'Spider-man'

    beforeEach(() => {
        wrapper = shallow(<SearchForm findNews={mockHeadline} />);
    })

    it('should match snapshot',() => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should check if state has changed when invoking handleChange',() => {
        let mockState = {
            target: {
                name: 'title',
                value: 'Spidey senses'
            }
        };

        let expected = 'Spidey senses';

        wrapper.instance().handleChange(mockState);

        expect(wrapper.state('title')).toEqual(expected);
    });

    it('should clear out state', () => {
      //setup

      let mockState = {
          headline: 'health'
      }

      let expected = {
          headline: ''
      }
      //execution

      wrapper.instance().setState(mockState);
      wrapper.instance().emptyInputs();

      //expectation
      expect(wrapper.state()).toEqual(expected);
    });

    it('should invoke both findNews and emptyInputs when findArticle is called',() => {

        const mockEvent = { preventDefault: jest.fn()};
        wrapper.instance().emptyInputs = jest.fn();
        const mockHeadline = { headline: 'title'};

        wrapper.instance().findArticle(mockEvent);

        expect(mockFindArticle).toHaveBeenCalledWith(mockHeadline);
        expect(wrapper.instance().emptyInputs).toHaveBeenCalled();
    });

    it('should invoke findArticle on click ', () => {
        wrapper.instance.findArticle = jest.fn();
        // wrapper.instance.forceUpdate();
        const mockEvent = { preventDefault: jest.fn() };
        wrapper.find('button').simulate('click', mockEvent);

        expect(wrapper.instance().findArticle).toHaveBeenCalledWith();
    })
})