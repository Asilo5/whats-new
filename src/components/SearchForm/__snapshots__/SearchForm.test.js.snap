// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<form>
  <input
    className="search-input"
    name="headline"
    onChange={[Function]}
    placeholder="Search for news article here."
    type="text"
    value=""
  />
  <button
    className="search-button"
    onClick={[Function]}
  >
     Search Now 
  </button>
</form>
`;
